src/inttau2.f90

```diff
@@ -252,20 +260,21 @@ subroutine update_jmean(grid, pos, dir, d_sdf, packet, mua)
             if(d + dcell > d_sdf)then
                 dcell = d_sdf - d
                 d = d_sdf
-! needs to be atomic so don't write to same array address with more than 1 thread at a time
+! needs to be atomic so dont write to same array address with more than 1 thread at a time
+                    packet%phase = packet%phase + dcell
+                    phasec = packet%energy*cmplx(cos(packet%fact*packet%phase), sin(packet%fact*packet%phase))
 !$omp atomic
-                    jmean(celli, cellj, cellk) = jmean(celli, cellj, cellk) + dcell
-!$omp atomic
-                    absorb(celli, cellj, cellk) = absorb(celli, cellj, cellk) + dcell*mua_real
+                    phasor(celli,cellj,cellk) = phasor(celli,cellj,cellk) + phasec!*mua_real
+                    !jmean(celli, cellj, cellk) = jmean(celli, cellj, cellk) + dcell*mua_real
```

Comment:
Please re-add all jmeans and absorbs here. Make sure that !$omp atomic is before each one though